### PR TITLE
Flink: port table refresh (#8555) to v1.15 and v1.16

### PR DESCRIPTION
**Pull request by bryanck**
This PR ports the changes from #8555 (optional table refresh) to Flink version 1.15 and 1.16.

This includes two minor updates for all versions, that were missed in the original PR:
* typo fix `TABLE_REFRSH_INTERVAL` -> `TABLE_REFRESH_INTERVAL`
* import fix `org.apache.calcite.linq4j.function.Experimental` -> `org.apache.flink.annotation.Experimental`

### CODE REVIEW

File: flink/v1.17/flink/src/test/java/org/apache/iceberg/flink/sink/TestFlinkIcebergSink.java [hunk 1]
```diff
@@ -379,7 +379,7 @@ public void testWriteRowWithTableRefreshInterval() throws Exception {
             .map(CONVERTER::toInternal, FlinkCompatibilityUtil.toTypeInfo(SimpleDataUtil.ROW_TYPE));
 
     Configuration flinkConf = new Configuration();
-    flinkConf.setString(FlinkWriteOptions.TABLE_REFRSH_INTERVAL.key(), "100ms");
+    flinkConf.setString(FlinkWriteOptions.TABLE_REFRESH_INTERVAL.key(), "100ms");
```
Comment (amogh-jahagirdar) [inline flink/v1.17/flink/src/test/java/org/apache/iceberg/flink/sink/TestFlinkIcebergSink.java:382]: Thanks for fixing this!

**Review by amogh-jahagirdar — APPROVED** [review]
Looks good to me!

File: flink/v1.15/flink/src/main/java/org/apache/iceberg/flink/FlinkWriteOptions.java [hunk 2]
```diff
@@ -18,6 +18,8 @@
  */
 package org.apache.iceberg.flink;
 
+import java.time.Duration;
+import org.apache.calcite.linq4j.function.Experimental;
```
Comment (nastra) [inline flink/v1.15/flink/src/main/java/org/apache/iceberg/flink/FlinkWriteOptions.java:22]: looks like the wrong import is being used here

File: flink/v1.15/flink/src/main/java/org/apache/iceberg/flink/FlinkWriteConf.java [hunk 3]
```diff
@@ -18,7 +18,9 @@
  */
 package org.apache.iceberg.flink;
 
+import java.time.Duration;
 import java.util.Map;
+import org.apache.calcite.linq4j.function.Experimental;
```
Comment (nastra) [inline flink/v1.15/flink/src/main/java/org/apache/iceberg/flink/FlinkWriteConf.java:23]: wrong import?

Comment (nastra) [reply flink/v1.15/flink/src/main/java/org/apache/iceberg/flink/FlinkWriteConf.java:23]: if we fix this one we should also fix it for flink 1.17

**Review by nastra — APPROVED** [review]
LGTM, but I added a comment about a (potentially) wrong import on the `Experimental` usage

File: flink/v1.15/flink/src/main/java/org/apache/iceberg/flink/FlinkWriteConf.java (hunk 3 above)
Comment (bryanck) [reply flink/v1.15/flink/src/main/java/org/apache/iceberg/flink/FlinkWriteConf.java]: Hm, apologies, I fixed these.

**Comment by pvary** [comment]
@bryanck: is this a clean backport? We tend to fix, change everything in 1.17 code , and then do a clean backport to 1.16, 1.15 whenever possible.  If there is some change, we highlight it in comments on github.

**Comment by bryanck** [comment]
@pvary This is a clean backport. Two minor items were discovered on this review which were applied to all versions, I added those to the description.

**Review by Fokko — APPROVED** [review]

**Review by nastra — APPROVED** [review]
LGTM, thanks @bryanck. Also thanks to @Fokko / @amogh-jahagirdar / @pvary for reviewing

**Comment by pvary** [comment]
> @pvary This is a clean backport. Two minor items were discovered on this review which were applied to all versions, I added those to the description.

Thanks!
Well done!

**Pull request merged**